Code/Framework/AzCore/AzCore/std/hash.h

```diff
@@ -34,7 +34,14 @@ namespace AZStd
     {
         typedef T               argument_type;
         typedef AZStd::size_t   result_type;
-        constexpr result_type operator()(const argument_type& value) const { return static_cast<result_type>(value); }
+        // If you implement your own hash operator, consider making it constexpr if it makes sense.
+        // This is intentionally not declared constexpr here, as this would force every implementation
+        // to require constexpr or suffer a compile error in certain compilers.  This means forcing them to implement
+        // their hash function in their headers, without being able to forward declare them.
+        // If you end up in a situation where you have implemented a custom constexpr operator() for your custom type
+        // and find that it needs to call an existing hash::operator() that is auto-generated from the below non-constexpr,
+        // you can add an explicit instantiation for it with constexpr.  See the example of hash<AZ::u64> below.
+        result_type operator()(const argument_type& value) const { return static_cast<result_type>(value); }
```

Comment:
Now don't we have speicializations for all the arithmetic types, we don't actually need to have the base template provide an `operator()` function at all.

Reply:
We don't have to do that now.
This can be done at a later point as a standalone PR>

Reply:
removing it leads to many many instances where it fails

eg. Hasher=AZStd::hash<AzNetworking::ConnectionId>,
Hasher=AZStd::hash<AZ::SliceComponent::SliceInstance>,
Hasher=AZStd::hash<AzNetworking::SocketFd>,
Hasher=AZStd::hash<AzNetworking::PacketId>,
Hasher=AZStd::hash<AzNetworking::SequenceId>,
and many more.

For now, we can keep it, nothing to gain by removing it here.